### PR TITLE
Limit force shadow update to urgent blocks

### PR DESCRIPTION
**Pull request by x2048**
Significantly improves performance when:
* shadows are enabled
* shadow_update_frames > 1
* mesh generation is active (world is loading, camera changes etc.)

## To do

This PR is Ready for Review.

## How to test

1. Enable shadows
2. Set shadow_update_frames to a value > 1 (default: 8)
3. Enter the world
4. Observe the graphs: mesh generation should not have significant impact on FPS
5. Place a block / dig a block - shadow is updated without lag.

### CODE REVIEW

File: src/client/mesh_generator_thread.cpp [hunk 1]
```diff
@@ -149,6 +149,7 @@ QueuedMeshUpdate *MeshUpdateQueue::pop()
 		m_queue.erase(i);
 		m_urgents.erase(q->p);
 		fillDataFromMapBlockCache(q);
+		q->urgent = must_be_urgent;
```
Comment (sfan5) [inline src/client/mesh_generator_thread.cpp:152]: shouldn't this be set at the point where's it put into the queue?

Comment (x2048) [reply src/client/mesh_generator_thread.cpp]: Moved there.

**Comment by lhofhansl** [comment]
Looks good. (But have not tested, yet)

**Review by sfan5 — APPROVED** [review]
Confirmed that the hit count (not value) of `SHADOW MapBlocks drawn [#]` is significantly lower while flying through newly generated world

**Comment by lhofhansl** [comment]
@x2048 It seems on camera rotation we should immediately recalculate the shadows. Right when you turns towards a shaded area you can see that the shadow is first faint and then fills in (after shadow_update_frames frames).

**Pull request merged**